nemo/collections/nlp/data/dialogue/dataset/dialogue_bert_dataset.py

```diff
@@ -116,8 +158,103 @@ def __getitem__(self, idx):
             np.array(self.all_subtokens_mask[idx]),
             self.all_intents[idx],
             np.array(self.all_slots[idx]),
+            np.array(self.all_bio_slots[idx]),
+            np.array(self.bio_mention_labels[idx]),
+            np.array(self.mention_loss_mask[idx]),
         )
 
+    @staticmethod
+    def get_bio_slot_label_from_sequence(slot_label_list, label_of_other_type):
+        """
+        Generate BIO slot label based on slot label
+        Args:
+            slot_label_list: list of int representing slot class of each word token (per sentence)
+            Eg,
+            send me a  wake up alert at seven am tomorrow morning
+            54   54 54 0    0  54    54 46    46 12       48
+            [54, 54, 54, 0, 0, 54, 54, 46, 46, 12, 48]
+
+            label_of_other_type: int lable of "Other" type in the slot_label_list
+            For instance, in drive_through dataset "Other" is 0; in assistant dataset "Other" is 54
+        Returns:
+            bio_label_list: list of int representing BIO slot class of each word token
+            eg,
+            send me a  wake up alert at seven am tomorrow morning
+            0    0  0  1    2  0     0  1     2  1        1
+            [0, 0, 0, 1, 2, 0, 0, 1, 2, 1, 1]
+        """
+
+        bio_label_list = []
+        start = 0
+        idx = 0
+
+        while idx < len(slot_label_list):
```

Comment:
I think this logic can be improved to make it shorter and more readable. Can you think about how this can be done?

Reply:
Pair coding with @Zhilin123  and solved it.